br/pkg/lightning/common/retry.go

```diff
@@ -101,11 +102,18 @@ func isSingleRetryableError(err error) bool {
 		if nerr.Timeout() {
 			return true
 		}
-		if cause, ok := nerr.(*net.OpError); ok {
-			syscallErr, ok := cause.Unwrap().(*os.SyscallError)
-			if ok {
-				return syscallErr.Err == syscall.ECONNREFUSED || syscallErr.Err == syscall.ECONNRESET
-			}
+		var (
+			syscallErr *os.SyscallError
+			ok         bool
+		)
+		switch cause := nerr.(type) {
+		case *net.OpError:
+			syscallErr, ok = cause.Unwrap().(*os.SyscallError)
+		case *url.Error:
+			syscallErr, ok = cause.Unwrap().(*os.SyscallError)
```

Comment:
`errors.Unwrap(nerr)`?

Reply:
addressed in https://github.com/pingcap/tidb/pull/43432/commits/c59843295b7cc9e9e9bc3f783140cb595175fd93